src/components/Table/Header.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { ReactElement } from 'react'
+import { HeaderGroup, CellValue } from 'react-table'
+import classNames from 'classnames'
+import { Icon, Text } from '../'
+import { calcColumnWidth, CHECKBOX_DEFAULT_WIDTH, CHECKBOX_HEADER_ID } from './utils'
+
+type Props = {
+  fixedHeader?: boolean
+  withSelect: boolean
+  tableWidth: number
+  headerGroup: HeaderGroup
+}
+
+function Header({ headerGroup, tableWidth, withSelect, fixedHeader = false }: Props): ReactElement {
+  return (
+    <tr
+      {...headerGroup.getHeaderGroupProps()}
+      className={classNames({ fixed_header: fixedHeader })}
+    >
+      {headerGroup.headers.map(
+        (
+          {
+            id,
+            minWidth,
+            maxWidth,
+            width,
+            widthInPercent,
+            getHeaderProps,
+            fixed,
+            columnProps,
+            render,
+            isSorted,
+            isSortedDesc
+          }: CellValue,
+          i: number,
+          arr: CellValue[]
+        ) => {
+          const isSelection = id === CHECKBOX_HEADER_ID
+
+          const style = {
+            width: isSelection ? 17 : calcColumnWidth(widthInPercent, tableWidth),
+            left: !isSelection && withSelect && fixed === 'left' ? CHECKBOX_DEFAULT_WIDTH : 0,
+            ...(!isSelection && minWidth ? { minWidth } : {}),
+            ...(!isSelection && maxWidth ? { maxWidth } : {}),
+            ...(!isSelection && width ? { width } : {})
```

Comment:
can't we merge ...(!isSelection && width ? { width } : {}) with line 41?

Reply:
this is more clear to understand that in case if we have width attribute it's overriding widthInPercent prop.